src/rpc/rawtransaction.cpp

```diff
@@ -1470,6 +1470,139 @@ UniValue sendrawtransaction(const UniValue &params, bool fHelp)
     return hashTx.GetHex();
 }
 
+void InputDebuggerToJSON(const CInputDebugger &input, UniValue &result)
+{
+    std::map<std::string, std::string>::iterator it;
+
+    result.pushKV("isValid", input.isValid);
+    UniValue uv_vdata(UniValue::VARR);
+    for (auto data : input.vData)
+    {
+        UniValue entry(UniValue::VOBJ);
+        entry.pushKV("isValid", data.isValid);
+        UniValue entry_metadata(UniValue::VOBJ);
+        for (it = data.metadata.begin(); it != data.metadata.end(); ++it)
+        {
+            entry_metadata.pushKV(it->first, it->second);
+        }
+        entry.pushKV("metadata", entry_metadata);
+        UniValue entry_errors(UniValue::VARR);
+        for (auto error : data.errors)
+        {
+            entry_errors.push_back(error);
+        }
+        entry.pushKV("errors", entry_errors);
+        uv_vdata.push_back(entry);
+    }
+    result.pushKV("inputs", uv_vdata);
+}
+
+UniValue validaterawtransaction(const UniValue &params, bool fHelp)
+{
+    if (fHelp || params.size() < 1 || params.size() > 3)
+    {
+        throw std::runtime_error(
+            "validaterawtransaction \"hexstring\" ( allowhighfees, allownonstandard )\n"
+            "\nValidates raw transaction (serialized, hex-encoded) to local node without broadcasting it.\n"
+            "\nAlso see createrawtransaction and signrawtransaction calls.\n"
+            "\nArguments:\n"
+            "1. \"hexstring\"    (string, required) The hex string of the raw transaction)\n"
+            "2. allowhighfees    (boolean, optional, default=false) Allow high fees\n"
+            "3. allownonstandard (string 'standard', 'nonstandard', 'default', optional, default='default')\n"
+            "                    Force standard or nonstandard transaction check\n"
+            "\nResult:\n"
+            "\"hex\"             (string) The transaction hash in hex\n"
```

Comment:
result needs to be properly specified -- I think what's there now is a cut/paste leftover

Reply:
fixed